### PR TITLE
Refreshing Contributing Guidelines

### PR DESCRIPTION
**Pull request by arielkr256**
### Background

Refreshes contribution guidelines in preparation for panther-analysis seasonal newsletter.

### Changes

- Refresh CONTRIBUTING.md
- Add STYLE_GUIDE.md

### CODE REVIEW

File: STYLE_GUIDE.md
Comment (cara-panther) [inline STYLE_GUIDE.md:8]: typo: identifier

File: STYLE_GUIDE.md
Comment (cara-panther) [inline STYLE_GUIDE.md:7]: sugg: 
- RuleID, Filename, and DisplayName should all be similar to one another. A good litmus test is: If you have the RuleID, would you be able to identify the related Python file (its Filename), and vice versa?

File: STYLE_GUIDE.md
Comment (cara-panther) [inline STYLE_GUIDE.md:9]: add "Example:" above the codeblock

File: STYLE_GUIDE.md
Comment (cara-panther) [inline STYLE_GUIDE.md:22]: The Reference value should be a link to a relevant security or threat research report that describes the attack this rule detects, as well as why it is valuable to detect it, from a security perspective. Avoid generic documentation links, such as general API or log source pages.

File: STYLE_GUIDE.md
Comment (cara-panther) [inline STYLE_GUIDE.md:30]: saying tags here is a little confusing since Tags is a different detection value you mention just after this. is there another word we can use instead? IDs?

File: STYLE_GUIDE.md
Comment (cara-panther) [inline STYLE_GUIDE.md:33]: add "Example:"

File: STYLE_GUIDE.md
Comment (cara-panther) [inline STYLE_GUIDE.md:28]: you could consider making this "Reports" since the report name can be anything I think, but I also understand that Mitre is the most commonly used

File: STYLE_GUIDE.md
Comment (cara-panther) [inline STYLE_GUIDE.md:62]: Use instead of Using

File: STYLE_GUIDE.md
Comment (cara-panther) [inline STYLE_GUIDE.md:81]: Use instead of Using

File: CONTRIBUTING.md
Comment (cara-panther) [inline CONTRIBUTING.md:3]: aww this is nice. sugg update to second sentence: "We appreciate all types of contributions, including new detection rules, feature requests, and bug reports."

File: CONTRIBUTING.md
Comment (cara-panther) [inline CONTRIBUTING.md:5]: big nit: in the docs, we use sentence case for section headers (and title case for page titles). so sentence case for this header would be: What makes a good detection?

I won't comment on all the headers on these pages, but same feedback applies

File: CONTRIBUTING.md
Comment (cara-panther) [inline CONTRIBUTING.md:7]: sugg: end first sentence (after rules) with a colon (:)

and put the second sentence after the list of resources

File: CONTRIBUTING.md
Comment (cara-panther) [inline CONTRIBUTING.md:9]: "Jack" is a little too familiar. sugg: The blog post Panther's founder, Jack Naglieri, wrote on [The ...

File: CONTRIBUTING.md
Comment (cara-panther) [inline CONTRIBUTING.md:21]: can you move this codeblock under the last bullet point above? adding: "to ensure CI tests pass, using the following commands:"

File: CONTRIBUTING.md
Comment (cara-panther) [inline CONTRIBUTING.md:28]: I don't think you updated step 1 in Pull Request Process, but sugg change to:
1. Make desired detection changes. This may include creating new detections in existing log type directories, creating new log type directories, updating existing detections, etc.

File: CONTRIBUTING.md
Comment (cara-panther) [inline CONTRIBUTING.md:33]: also don't think this changed here, but on #5, sugg change to:
5. Once your PR has been approved by code owners, if you have merge permissions, merge it. If you do not have merge permissions, leave a comment requesting a code owner merge it for you.

File: CONTRIBUTING.md
Comment (cara-panther) [inline CONTRIBUTING.md:38]: nit: "this" project

**Review by le4ker — APPROVED** [review]

**Review by remy-says-hi — COMMENTED** [review]
Hey! Just realized I did not submit. Thank you for putting these changes together, Ariel. This is exactly what we need!

**Pull request merged**
